pkg/services/accesscontrol/scope_test.go

```diff
@@ -22,34 +23,156 @@ func TestResolveKeywordedScope(t *testing.T) {
 		name       string
 		user       *models.SignedInUser
 		permission Permission
-		want       *Permission
+		want       Permission
 		wantErr    bool
 	}{
 		{
 			name:       "no scope",
 			user:       testUser,
 			permission: Permission{Action: "users:read"},
-			want:       &Permission{Action: "users:read"},
+			want:       Permission{Action: "users:read"},
 			wantErr:    false,
 		},
 		{
 			name:       "user if resolution",
 			user:       testUser,
 			permission: Permission{Action: "users:read", Scope: "users:self"},
-			want:       &Permission{Action: "users:read", Scope: "users:id:2"},
+			want:       Permission{Action: "users:read", Scope: "users:id:2"},
 			wantErr:    false,
 		},
 	}
 	for _, tt := range tests {
+		var err error
 		t.Run(tt.name, func(t *testing.T) {
 			resolver := NewScopeResolver()
-			resolved, err := resolver.ResolveKeyword(tt.user, tt.permission)
+			scopeModifier := resolver.GetResolveKeywordScopeMutator(tt.user)
+			tt.permission.Scope, err = scopeModifier(context.TODO(), tt.permission.Scope)
 			if tt.wantErr {
 				assert.Error(t, err, "expected an error during the resolution of the scope")
 				return
 			}
 			assert.NoError(t, err)
-			assert.EqualValues(t, tt.want, resolved, "permission did not match expected resolution")
+			assert.EqualValues(t, tt.want, tt.permission, "permission did not match expected resolution")
+		})
+	}
+}
+
+func TestScopeResolver_ResolveAttribute(t *testing.T) {
```

Comment:
> scope pun intended

:laughing: 

> question: does the attribute scope resolver work with wildcards? If so, it'd be good to add a test with something like

Good question, my thought was that it would make sense to use `datasources:name:*` since it's `datasources:id:*` but it might be worth covering the case... Just in case ?

Reply:
I think you're right that resolving a wildcard to a wildcard would work logic wise. Not sure whether the code currently supports it - it feels like we'd try to find a datasource with name `*`? But I might be reading it wrong 🤔  Either way would be good to add a test for it.

Reply:
Sorry I wasn't clear, I was stating the code doesn't support wildcard, because it doesn't really make sense for someone to use an evaluator with `resource:name:*` they should use `resource:id:*` straight away.
But if we think it would still make sense, we can cover this extra case.

Reply:
Ah, gotcha! And covering it sounds good, as I think it wouldn't be too much extra work. But I also see your point of just using the ID based scope it that case. So I think go with the approach that you think is clearer and more readable :)

Reply:
Updated the datasource name solver accordingly